### DISCUSSION TITLE
Can catalog support publish multiple .toml files to a single catalog-plugin?

### DISCUSSION THREAD

**Discussion by xiaobailong24** [telereso/kmp-core#38] [Ideas]
There are three versions.toml:
- kotlin.versions.toml
- google.versions.toml
- third.versions.toml

How to publish them to a single plugin?
Except to use:
```
dependencyResolutionManagement {
    repositories {
        google()
        mavenCentral()
    }
    versionCatalogs {
        create("kmpLibsKotlin") { 
            from("io.telereso.kmp:catalog:0.17")
        }
        create("kmpLibsGoogle") { 
            from("io.telereso.kmp:catalog:0.17")
        }
        create("kmpLibsThird") { 
            from("io.telereso.kmp:catalog:0.17")
        }
    }
}
```

**Comment by ahna92** [comment]
hi @xiaobailong24 
sorry about the late replay 

I think the only way would be to build the catalog with code-base approach https://docs.gradle.org/current/userguide/platforms.html#sub:version-catalog-declaration  

you can see I'm doing something similar [here](https://github.com/telereso/kmp-core/blob/main/catalog/build.gradle.kts)

so your project will have 3 plugins , then combine into one final one

but i feel it's more overhead to maintain , but up to you 😄
